job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala

```diff
@@ -80,8 +82,8 @@ class LocalContextSupervisorSpec extends TestKit(LocalContextSupervisorSpec.syst
 
     it("should be able to add multiple new contexts") {
       supervisor ! AddContext("c1", contextConfig)
-      supervisor ! AddContext("c2", contextConfig)
       expectMsg(ContextInitialized)
+      supervisor ! AddContext("c2", contextConfig)
```

Comment:
I suspect this will take a while to investigate.  I'd like to understand
further why it doesn't work......   as a first pass, how about posting the
logs from the test log when you don't switch the order of the
initialization?

Serializing is not hard, we set up a queue and make it event-based; ie when
one gets an event from the context that the initialization is done, then we
can create the second context.   Let's figure out though why first.  One
initial problem was that creating multiple contexts at once was not
possible due to global environment vars, but that problem has been solved
already.

On Fri, Jan 16, 2015 at 7:23 AM, Fernando Otero <notifications@github.com>
wrote:

> In job-server/test/spark.jobserver/LocalContextSupervisorSpec.scala
> <https://github.com/spark-jobserver/spark-jobserver/pull/39#discussion-diff-23085865>
> :
>
> >        expectMsg(ContextInitialized)
> > +      supervisor ! AddContext("c2", contextConfig)
>
> I included this change just to talk about it, since it actually seems like
> there's a race condition, I had no intentio to merge this change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spark-jobserver/spark-jobserver/pull/39/files#r23085865>
> .
>



-- 
The fruit of silence is prayer;
the fruit of prayer is faith;
the fruit of faith is love;
the fruit of love is service;
the fruit of service is peace.  -- Mother Teresa